firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionLifecycleService.kt

```diff
@@ -142,7 +144,37 @@ internal class SessionLifecycleService() : Service() {
     }
 
     /** Generates a new session id and sends it everywhere it's needed */
-    private fun newSession() {
+    private suspend fun newSession() {
+      val subscribers = FirebaseSessionsDependencies.getRegisteredSubscribers()
+
+      if (subscribers.isEmpty()) {
+        Log.d(
+          TAG,
+          "Sessions SDK did not have any dependent SDKs register as dependencies. Events will not be sent."
+        )
+        return
+      }
+
+      if (subscribers.values.none { it.isDataCollectionEnabled }) {
+        Log.d(TAG, "Data Collection is disabled for all subscribers. Skipping this Session Event")
+        return
+      }
+
+      Log.d(TAG, "Data Collection is enabled for at least one Subscriber")
+
+      // This will cause remote settings to be fetched if the cache is expired.
+      SessionsSettings.instance.updateSettings()
```

Comment:
I think we should move this logic to the FirelogUploader and just prevent the upload there.

Reply:
Done